src/org/labkey/test/components/ui/grids/ResponsiveGrid.java

```diff
@@ -644,6 +644,19 @@ public List<String> getGridMessages()
                 .child(Locator.tagWithClass("div", "grid-message")).findElements(this));
     }
 
+    public boolean hasGridError()
+    {
+        return getGridError() != null;
+    }
+
+    public String getGridError()
+    {
+        Locator.XPathLocator errorLocator = Locator.tagWithClass("div", "alert-danger");
+        if (errorLocator.existsIn(this))
+            return errorLocator.findElement(this).getText();
+        return null;
```

Comment:
Could do something like this to avoid finding the element multiple times.
```suggestion
        Locator.XPathLocator errorLocator = Locator.tagWithClass("div", "alert-danger");
        return errorLocator.findOptionalElement(this).map(WebElement::getText).orElse(null);
```

Reply:
done